### ISSUE TITLE
[0.53] Commits to cherry-pick into stable release

### ISSUE THREAD

**Issue by grabbou** [facebook/react-native#17520]
The thread containing issues that are blocking the 0.53 release and need to be addressed.

Please keep this list to critical bug fixes that are new in 0.53 and ideally have been merged into master -- post commit hashes rather than issue/PR numbers unless the issue/PR in question targets a problem specific to 0.53. The intent of these cherry-pick threads is not to solicit features but to fix things that make 0.53 substantially more launch-blocked than 0.52.

**Comment by ide** [comment]
Cherry-picked 8b64907eb4542a74933929c43fdd5569357f5f54 (fix HMR client message) for CRNA

**Comment by fernandopasik** [comment]
I think you'll need to cherry-pick https://github.com/facebook/react-native/commit/ee5c91c0317b0defbb8da21f7e6d8d3ac8967381 as well as it fixes the CI on the android tests

**Comment by Andreyco** [comment]
Unify `boost` vs `boost-for-react-native` dependency, commit here d40db3a715afaf1cde4a5e231e96e46b2808bbef
Greatly improves build times.

**Comment by vonovak** [comment]
this is a bugfix for a RefreshControl issue introduced in 0.51 (or 0.52?), would be nice to have it fixed already in 0.53: https://github.com/facebook/react-native/commit/ed5872e2cca955ee407e87e37e13c7fed182199a

**Comment by ide** [comment]
Cherry-picked in the RN Boost and RefreshControl commits. The Yoga commit does not merge cleanly and affects code that landed after 0.53 was cut.

**Comment by grabbou** [comment]
Just FYI, I'll be releasing stable version tomorrow in case you have any
remaining commits to cherry pick.

**Comment by ide** [comment]
@grabbou I think that’s everything for now in 0.53-stable.

**Comment by brunolemos** [comment]
Is there a thread for non critical commits?

---
Context: The PR #16933 got merged to master (2-line change), but the import was messed up and deleted the whole file (https://github.com/facebook/react-native/commit/0d4ff1b7ea768cceca0405c665e322c0d6b5ba20) and then it was recreated in another commit (https://github.com/facebook/react-native/commit/b8c86b8deced01027b609959576ffcf5d2d0f520) which does include the PR mentioned but can easily go unnoticed, so that's why I'm mentioning it here.

**Comment by grabbou** [comment]
No, we don't @brunolemos. Best to just let me know around the end / beginning of a month when I am about to release a new version.

**Comment by grabbou** [comment]
Meanwhile, let me close this issue as I am promoting 0.53 to stable.

**Comment by jozan** [comment]
I found one bug and fixed it in local-cli. It prevents us from upgrading since it breaks running on actual iOS device. I was directed to try pinging this thread in hopes to get patch release.

PR in question: https://github.com/facebook/react-native/pull/17983

**Comment by jozan** [comment]
Commit: https://github.com/facebook/react-native/commit/f8fee0a631d77313d7cb5e65a3dd04a5a8ba3d03

Sorry for ugly description, I don’t know how all this works...

**Comment by MrLoh** [comment]
It’s very unfortunate that 0.53 was release while not being stable. Without being able to run on device this is not very usable. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by jozan** [comment]
It’s is unfortunate but not unheard of and definitely not end of the world.
Bug are part of software anyways. Creating a fork is regular procedure and
in some cases even necessary. 0.53 is really great update with lots of
other fixes. It didn’t work for us out of the box but it’s too big of a
deal.

**Comment by MrLoh** [comment]
Yeah I really appreciate the effort of all the people on making React native work. And 0.53 is a great release. And of course bugs always happen. Developing react native for around 6 months now, I’m just sometimes really frustrated how unstable everything still is (of course it’s beta) but It’s such a popular option for production apps. Whenever I do some web development now I’m amazed by how incredibly stable things are in comparison. Obviously the task for React native is monumental and it will probably take years to reach a similar level of maturity as web development. So everyone who is working towards that has my huge appreciation, it’s sometimes just hard to cope with the frustration of spending many hours again on just getting things to work. I have no idea how to improve this either, I guess we just all have to help where we can and bring a lot more patience and appreciation for the complexity of the problem that react native solves. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by grabbou** [comment]
Thank you @MrLoh and @jozan for taking your time to share your thoughts on the latest stable release. I appreciate your feedback. I have cherry-picked this small fix to unblock you from upgrading. Should be out on npm within next couple of hours.

I agree that React Native has its unstable moments from time to time, but as others successfully pointed out, it is still a fairly new technology. The biggest problem with releases being sometimes prematurely promoted to stable is that the amount of people running on `RC` channel is limited. That makes it hard to spot problems before hitting stable channel.

We are doing our best to make that as pleasant experience as possible by controlling post-release state of issues and coordinating with Facebook the release status.

**Comment by MrLoh** [comment]
@grabbou thank you so much for making this release!! We are very grateful for all the hard work that you and so many others put into React Native. RN is a truly amazing technology solving a really hard problem and having a big impact on App development.
